### PR TITLE
Breakout RL experiments

### PR DESCRIPTION
**Pull request by dee0512**
Major changes: 

1. Make the softmax decision on "sum of spikes" using spike monitors. 
2. Add 2 new breakout experiments.

### CODE REVIEW

File: bindsnet/pipeline/__init__.py [hunk 1]
```diff
@@ -75,17 +75,20 @@ def __init__(self, network: Network, environment: Environment, encoding: Callabl
         else:
             self.history = {}
 
+        # monitor spikes for selecting action based on sum of spikes
```
Comment (djsaunde) [inline bindsnet/pipeline/__init__.py]: It seems like this shouldn't be done in the `Pipeline` object, but outside in an experimental script. If the layers of the network are really large, for example, this could make things really slow.

File: examples/breakout/breakout_stdp_right.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import torch
```
Comment (djsaunde) [inline examples/breakout/breakout_stdp_right.py]: What's the purpose of this script? We want to have only a few examples in the BindsNET repo so we don't have to maintain a bunch.

File: bindsnet/pipeline/__init__.py (hunk 1 above)
Comment (dee0512) [reply bindsnet/pipeline/__init__.py]: Yes. You are right. I will make the change.

File: examples/breakout/breakout_stdp_right.py (hunk 2 above)
Comment (dee0512) [reply examples/breakout/breakout_stdp_right.py]: @Hananel-Hazan wanted an experiment for reinforcing only the right action via STDP. This was to see if it performs well enough to include the results in the paper. At this point, I am not really sure which one we should use. The breakout.py example performs the best. Which is not saying much because it has no learning.

Comment (djsaunde) [reply examples/breakout/breakout_stdp_right.py]: Got it, thanks. It might be better to keep this script in the [bindsnet_experiments repo](https://github.com/djsaunde/bindsnet_experiments).

Comment (dee0512) [reply examples/breakout/breakout_stdp_right.py]: Ok. I will remove it for now. Since I am not sure about the utility of this experiment.

Comment (Hananel-Hazan) [reply examples/breakout/breakout_stdp_right.py]: It should be in directory with the rest of the benchmark scripts and example scripts that will appear in the paper for Frontier. @djsaunde will create a spacial tag for it
